### PR TITLE
[zksend] add support for bulk link creation, and simplified claim APIs

### PR DESCRIPTION
**Pull request by hayes-mysten**
## Description 

Describe the changes or additions included in this PR.

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and do not break anything, you can skip the following section. Otherwise, please briefly describe what has changed under the Release Notes section.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: sdk/zksend/src/links/builder.ts [hunk 1]
```diff
@@ -230,26 +222,26 @@ export class ZkSendLinkBuilder {
...
 		objectsToTransfer.push(gas);
 
-		txb.setSenderIfNotSet(this.#sender);
+		txb.setSenderIfNotSet(this.sender);
 		txb.transferObjects(objectsToTransfer, address);
 
 		return txb;
 	}
 
 	async #estimateClaimGasFee(): Promise<bigint> {
```
Comment (Jordan-Mysten) [inline sdk/zksend/src/links/builder.ts:236]: Just to confirm this is only used in the old link style right?

File: sdk/zksend/src/links/builder.ts [hunk 2]
```diff
@@ -284,11 +276,137 @@ export class ZkSendLinkBuilder {
...
+					type: res.data.type!,
+				});
+			}
+		}
+
+		const mergedCoins = new Map<string, TransactionObjectArgument>([
+			[SUI_COIN_TYPE, transactionBlock.gas],
+		]);
+
+		for (const [coinType, coins] of coinsByType) {
```
Comment (Jordan-Mysten) [inline sdk/zksend/src/links/builder.ts:360]: Can't wait for intents <3

File: sdk/zksend/src/links/list-created-links.ts [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+					... on ProgrammableTransactionBlock {
+						inputs(first: 10) {
+							nodes {
+								__typename
+								... on Pure {
+									bytes
+								}
+							}
+						}
+						transactions(first: 10) {
```
Comment (Jordan-Mysten) [inline sdk/zksend/src/links/list-created-links.ts:41]: Why only the first 10?

**Review by Jordan-Mysten — APPROVED** [review]

File: sdk/zksend/src/links/builder.ts (hunk 1 above)
Comment (hayes-mysten) [reply sdk/zksend/src/links/builder.ts:236]: correct

File: sdk/zksend/src/links/list-created-links.ts (hunk 3 above)
Comment (hayes-mysten) [reply sdk/zksend/src/links/list-created-links.ts:41]: 10 is currently the max, and 5 is the default, so this is the most we can get right now

**Pull request merged**